File: arrayDeletion.cpp
#include <iostream>
using namespace std;

void display(int arr[], int size){
    for (int i = 0; i < size; i++)
    {
        cout<<arr[i]<<" ";
    }
}
void deletion(int arr[], int *size, int index, int capacity){

    if (index > capacity)
    {
        cout<<"you exceded the capacity of array"<<endl;
    }
    if (index >= *size)
    {
        cout<<"you exceded the size of array"<<endl;
    }
    
    if (index < *size)
    {
        for (int i = index; i <= *size; i++)
        {
            arr[i] = arr[i+1];
        }
        (*size)--;
    }
}

int main(){
    int size = 5, capacity = 100, index;
    int arr[capacity] = {5, 6, 2, 8, 4};
    cout<<"enter the index on which you want deletion: ";
    cin>>index;
    deletion(arr, &size, index, capacity);
    cout<<"size of new array: "<<size<<endl;
    display(arr, size);
    return 0;
}

File: ArrayDeletion.java
import java.util.Scanner;

public class ArrayDeletion {
    public static int deletion(int arr[], int index, int size, int capacity){
        for (int i = index; i < size ; i++) {
            arr[i] = arr[i+1];
        }
        size--;
        return size;
    }
    
    public static void display(int arr[], int size) {
        for (int i = 0; i < size; i++) {
            System.out.printf("%d ", arr[i]);
        }
    }
    public static void main(String[] args) {
        int size = 5;
        int[] arr = new int[100];
        arr[0] = 32;
        arr[1] = 12;
        arr[2] = 2;
        arr[3] = 5;
        arr[4] = 20;
        int index; 
        System.out.println("enter the index you want to delete: ");
        Scanner sc = new Scanner(System.in);
        index = sc.nextInt();
        sc.close();
        size = deletion(arr, index, size, 100);
        display(arr, size);
    }
}
